store/src/java/com/zimbra/cs/volume/VolumeS3ConfigCLI.java

```diff
@@ -0,0 +1,526 @@
+import com.zimbra.soap.admin.message.DeleteS3BucketConfigRequest;
+import com.zimbra.soap.admin.message.DeleteS3BucketConfigResponse;
+import com.zimbra.soap.type.GlobalS3BucketConfiguration;
+import com.zimbra.soap.JaxbUtil;
+
+public final class VolumeS3ConfigCLI extends SoapCLI {
+    // cli constants
+    private static final String  O_A    = "a";
+    private static final String  O_D    = "d";
+    private static final String  O_L    = "l";
+    private static final String  O_BID  = "bid";
+    private static final String  O_BN   = "bn";
+    private static final String  O_AK   = "ak";
+    private static final String  O_SK   = "sk";
+    private static final String  O_DP   = "dp";
+    private static final String  O_R    = "r";
+    private static final String  O_ACC  = "acc";
+    private static final String  O_NS   = "ns";
+    private static final String  O_PP   = "pp";
+    private static final String  O_URL  = "url";
+    private static final String  O_SP   = "t";
+    // store provider constants
+    private static final String AWS_S3 = "AWS_S3";
+    private static final String CEPH_S3 = "CEPH_S3";
+    private static final String STORAGE_GRID_S3 = "CEPH_S3";
+    // variables
+    private ZAuthToken auth;
+    private String storeProvider = "";
+    private String bucketId = "";
+    private String bucketName = "";
+    private String accessKey = "";;
+    private String secretKey = "";
+    private String destPath = "";
+    private String region = "";
+    private String account = "";
+    private String namespace = "";
+    private String proxyPort = "";
+    private String accountPort = "";
+    private String url = "";
+
+    private VolumeS3ConfigCLI() throws ServiceException {
+        super();
+        setupCommandLineOptions();
+    }
+
+    @Override
+    protected void setupCommandLineOptions() {
+        super.setupCommandLineOptions();
+        // Main Option Group
+        OptionGroup og = new OptionGroup();
+        og.addOption(new Option(O_A, "add", false, "Adds a s3BucketConfig"));
+        og.addOption(new Option(O_D, "delete", false, "Deletes a s3BucketConfig"));
+        og.addOption(new Option(O_L, "list", false, "Lists all s3BucketConfig"));
+        og.setRequired(true);
+        // Options
+        Options options = getOptions();
+        options.addOptionGroup(og);
+        options.addOption(O_BN, "bucketName", true, "Specify name for s3BucketConfig");
+        options.addOption(O_AK, "accessKey", true, "Specify access key for s3BucketConfig");
+        options.addOption(O_SK, "secretKey", true, "Specify secret key for s3BucketConfig");
+        options.addOption(O_DP, "destPath", true, "Specify destination path for s3BucketConfig");
+        options.addOption(O_R, "region", true, "Specify region for s3BucketConfig");
+        options.addOption(O_URL, "url", true, "Specify url for s3BucketConfig");
+        options.addOption(O_SP, "type", true, "Specify storeProvider to add s3BucketConfig");
```

Comment:
it should be storageProvider right?

Reply:
should we keep storeProvider or storageProvider everywhere ?